storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -100,9 +103,13 @@ bool Redo_Log_Reader::find_start_checkpoint_lsn() {
     checkpoint_lsn_start = chkp_header.m_checkpoint_lsn;
     checkpoint_offset_start = chkp_header.m_checkpoint_offset;
 
+    debug_sync_point("stop_before_copy_log_hdr");
+
     /* Copy the header into log_hdr_buf */
     file_handle.read(0, LOG_FILE_HDR_SIZE, log_hdr_buf);
   }
+  /* update the headers to store the current checkpoint LSN */
+  update_log_temp_checkpoint(log_hdr_buf, checkpoint_lsn_start);
```

Comment:
without this call, how did we write checkpoint_lsn in redo file copied?

Reply:
we copied the hdr into log_hdr_buf. so that's how it copied checkpoint_LSN.

Reply:
Thanks, understood

Reply:
@rahulmalik87 this doesn't explain how it sets the correct m_file_start_lsn in the redo header?
@altmannmarcelo ^

Reply:
This is the place where we read the header file. and correct it by calling `update_log_temp_checkpoint`  during --backup.